src/Microsoft.AspNet.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -0,0 +1,678 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Text;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    /// <summary>
+    /// A media type value.
+    /// </summary>
+    public struct MediaType
+    {
+        private static readonly StringSegment QualityParameter = new StringSegment("q");
+
+        private readonly string _mediaType;
+        private readonly int _offset;
+        private readonly int? _length;
+        private Enumerator _parameterEnumerator;
```

Comment:
It's not, methods like GetParameter don't need to reparse type and subtype before they can search for the parameter, they can start at the right position.